kotest-framework/kotest-framework-engine/src/jvmTest/kotlin/com/sksamuel/kotest/engine/test/RetryTest.kt

```diff
@@ -0,0 +1,91 @@
+package com.sksamuel.kotest.engine.test
+
+import io.kotest.core.spec.style.DescribeSpec
+import io.kotest.core.spec.style.FunSpec
+import io.kotest.core.test.config.DefaultTestConfig
+import io.kotest.matchers.comparables.shouldBeGreaterThan
+import kotlin.time.Duration.Companion.milliseconds
+import kotlin.time.TimeSource
+
+class RetryTest : FunSpec() {
+   init {
+
+      retries = 3
+      retryDelay = 25.milliseconds
+
+      val timesource = TimeSource.Monotonic
```

Comment:
Currently, `TestCaseExecutor` uses the hardwired `TimeSource.Monotonic`, exposing us to flaky tests on CI. Could we change this to `testTimeSource()`, which would provide virtual time conditionally?

Then we could make this test stable by modifying it like so:

```suggestion
      coroutineTestScope = true
      val timesource = testTimeSource()
```